### ISSUE TITLE
Allow component to be marked as @internal

### ISSUE THREAD

**Issue by WhitWaldo** [dotnet/roslyn#85460]
This is technically a duplicate of both [#21213](https://github.com/dotnet/aspnetcore/issues/21213) and dotnet/razor#7051 but seeing as each of those are closed, I'm unable to provide my own feedback there, so I'm opening this new ticket.

I would like to request the ability to mark components with an internal accessibility. 

Today, I can ensure that any type in a project isn't accessible outside the project by marking it as internal. This is a useful means of ensuring that those types not intended for use outside of their current utility stay this way. An ASP.NET Core website implemented with Blazor is little different in that while no one _should_ ever be trying to reference the project as a means of accessing its various components, one might try to do so to access its various other types (e.g. POCOs, JSON converters and the like).

I went through my project and marked all my types as internal, but at the end of that effort ran into the issue that there appears no way of marking components the same. This is bad, as this means I've run into the issue where my generically-implemented components are now more accessible than the types that they're constrained to giving me a build error.

For example, given a class like the following:

```cs
internal class GenericTableState<T> where T : IRecord
{
  //...
}
```

And then a component, `GenericTable.razor`, implementing this with a typeparam itself:

```cs
@typeparam TState where TState : GenericTableState<TItem>
```

Both of these get a red squiggly with the following error:
> Inconsistent accessibility: constraint type 'GenericTableState<TItem>' is less accessible than 'GenericTable<TState, TItem>`

Having a means of marking the component as internal would both ensure that the component cannot be used outside of the project (good coding practice) and allow the project to build with consistent accessibility. Rather, until this is made available, I'm going to have to roll back _all_ the accessibility modifiers throughout the rest of the project and set to public which will have resulted in significant lost time and effort.

I'd be perfectly fine with the `@visibility internal` syntax, though I'm afraid I can't think of a great situation in which I might ever need private unless I were doing something creative with inheritance and needed more of a `private protected`. Anyway, just having support for internal would be fine at this juncture. Thank you for your renewed consideration!

**Comment by ScarletKuro** [comment]
Same issue on the aspnetcore repo https://github.com/dotnet/aspnetcore/issues/60747

**Comment by ViRuSTriNiTy** [comment]
I agree with all the comments in this issue. This little feature missing is an issue for component authors. So, hopefully this gets implemented soon. 🤞

**Comment by Dona278** [comment]
Changed the area label, another issue with a goo reason linked (https://github.com/dotnet/roslyn/issues/85646) but no any feedback by the team right?
